### ISSUE TITLE
os download, os versions: should support device type aliases

### ISSUE THREAD

**Issue by pdcastro** [balena-io/balena-cli#2386]
```
$ balena os download nuc -o os.img --version menu
Getting device operating system for nuc
Error: No balenaOS versions found for device type 'nuc'.
Double-check the device type slug with 'balena devices supported'.

$ balena os download intel-nuc -o os.img --version menu
Getting device operating system for intel-nuc
? Select the OS version: (Use arrow keys)
❯ v2.83.18+rev1.prod (recommended)
...

$ balena os versions nuc
Error: No balenaOS versions found for device type 'nuc'.
Double-check the device type slug with 'balena devices supported'.

$ balena os versions intel-nuc
v2.83.18+rev1.prod (recommended)
...
```

I.e., 'nuc' is not supported, even though it is an alias for 'intel-nuc':

```
$ balena devices supported
SLUG                                   ALIASES                ARCH    NAME
intel-nuc                              nuc                    amd64   Intel NUC
...
```

In the case of `os versions`, the issue was introduced in CLI v12.52.0 (PR #2334).
In the case of `os download`, it pre-existed.

**Comment by pdcastro** [comment]
Connects-to: #2005, #2334

**Comment by pdcastro** [comment]
Resolved in CLI v12.55.4.
